File: CinemaReservations/CSharp/CinemaAllocations.Domain/NoPossibleAllocationsFound.cs
using System.Collections.Generic;

namespace CinemaAllocations.Domain
{
    public class NoPossibleAllocationsFound : SeatsAllocated
    {
        public NoPossibleAllocationsFound(int partyRequested) : base(new List<Seat>(), partyRequested)
        {
        }
    }
}

File: CinemaReservations/Java/CinemaAllocationsDomain/src/main/java/com/sdd/cinemaallocations/NoPossibleAllocationsFound.java
package com.sdd.cinemaallocations;

import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.Value;

import java.util.List;

@Value
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
public class NoPossibleAllocationsFound extends SeatsAllocated {

    public NoPossibleAllocationsFound(int partyRequested, List<Seat> reservedSeats) {
        super(partyRequested, reservedSeats);
    }
}